src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/ITemporaryStorageTextHandle.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Host;
+
+internal interface ITemporaryStorageTextHandle
```

Comment:
>ITemporaryStorageTextHandle [](http://example.com/codeflow?start=19&length=27)

nit: would it be useful for ITemporaryStorageTextHandle and ITemporaryStorageStreamHandle to derive from a common interface?

Reply:
It almost seems like there could just be a single

ITemporaryStorageHandle<TStorage> interface that returned TStorage from the Read* methods

Reply:
Maybe?  I just don't see it buying much .   This isn't a system we've needed to generalize.  It handles byte streams and encoded strings effectively.  I'm fine keeping it specific to those two needs.